src/math_utils.rs

```diff
@@ -25,6 +25,15 @@ pub fn isqrt(n: &BigInt) -> Result<BigInt, VirtualMachineError> {
     Ok(x)
 }
 
+/// Performs integer division between x and y; fails if x is not divisible by y.
+pub fn safe_div(x: &BigInt, y: &BigInt) -> Result<BigInt, VirtualMachineError> {
+    if !(x % y).is_zero() || y.is_zero() {
```

Comment:
You need to invert this check, otherwise you risk dividing by zero.
```suggestion
    if y.is_zero() || !(x % y).is_zero() {
```

Reply:
Resolved: 5eed780ae84ab1dbd34da6e7d94f154b619f1200

Reply:
resolved https://github.com/lambdaclass/cleopatra_cairo/pull/333/commits/5eed780ae84ab1dbd34da6e7d94f154b619f1200